invoice/src/main/java/org/killbill/billing/invoice/generator/UsageInvoiceItemGenerator.java

```diff
@@ -196,7 +196,7 @@ private void updatePerSubscriptionNextNotificationUsageDate(final UUID subscript
                     public boolean apply(final InvoiceItem input) {
                         if (input.getInvoiceItemType() == InvoiceItemType.USAGE) {
                             final Usage usage = knownUsage.get(input.getUsageName());
-                            return usage.getBillingMode() == BillingMode.IN_ARREAR;
+                            return usage != null && usage.getBillingMode() == BillingMode.IN_ARREAR;
```

Comment:
nit: Fix makes sense and is simple enough - just wondering if a log line would make sense, since this is a case where we previously invoiced some usage, which is not not referenced in the current set of billing events.

Reply:
I'm always very wary of adding logging in `invoice`, because if the condition is met, it will always be met for each invoice run in the future. This could be quite noisy for these occasions where the user is actually ok with it (we have the _problem_ today in some Janitor GET paths in `payment`).